### PR TITLE
feat: add pbjson bytes support

### PR DESCRIPTION
**Pull request by tustvold**
The [JSON mapping](https://developers.google.com/protocol-buffers/docs/proto3#json) defines that bytes fields should be encoded/decoded as base64. This implements that functionality.

Unfortunately there is an outstanding bug in prost that prevents bytes from functioning correctly in maps (see https://github.com/tokio-rs/prost/issues/531) and so I've left out support for that admittedly niche use-case for now

### CODE REVIEW

File: pbjson_test/src/lib.rs [hunk 1]
```diff
@@ -197,12 +197,30 @@ mod tests {
         verify_decode(&decoded, "{}");
 
         // Bytes support currently broken
```
Comment (crepererum) [inline pbjson_test/src/lib.rs:199]:
```suggestion
```

File: pbjson/src/lib.rs [hunk 2]
```diff
@@ -38,4 +42,41 @@ pub mod private {
...
+    impl<'de, T> Deserialize<'de> for BytesDeserialize<T>
+    where
+        T: From<Vec<u8>>,
+    {
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: serde::Deserializer<'de>,
+        {
+            let str: &str = Deserialize::deserialize(deserializer)?;
+            let decoded = base64::decode(str).map_err(serde::de::Error::custom)?;
```
Comment (crepererum) [inline pbjson/src/lib.rs:58]:
```suggestion
            let s: &str = Deserialize::deserialize(deserializer)?;
            let decoded = base64::decode(s).map_err(serde::de::Error::custom)?;
```

Could we please not shadow builtins or use types as variable names?

**Review by crepererum — APPROVED** [review]
Looks good, at least as far as I can judge the code :D

File: pbjson_build/src/generator/message.rs [hunk 3]
```diff
@@ -717,6 +728,10 @@ fn write_deserialize_field<W: Write>(
                     )?;
                     "v.0"
                 }
+                FieldType::Scalar(ScalarType::Bytes) => {
+                    // https://github.com/tokio-rs/prost/issues/531
```
Comment (alamb) [inline pbjson_build/src/generator/message.rs:732]: 👍

**Review by alamb — APPROVED** [review]

**Pull request merged**